### DISCUSSION TITLE
Keybase meeting Thursday, March 18th 21h00 UTC

### DISCUSSION THREAD

**Discussion by huey735** [bisq-network/bisq#5322] [General]
Prompted by threads like https://github.com/bisq-network/bisq/discussions/5271 and a lack of a clear general direction for Bisq I decided to propose a call so that the more invested contributors can talk and better get a sense of Bisq's goals.

- Where we are going?
- What are the priorities?
- What isn't worth spending time on given where we're going?

A text meeting was suggested instead to be more inclusive so we are going to run a text-base-meeting on the `#team-meeting` channel on [Keybase])https://keybase.io/team/bisq) on Thursday, March 18th 21h00 UTC. The time is to be determined.

This is a new format for me so I'd like to establish some broad rules
- The person typing has the floor. If you see someone typing, do not type. You can start typing elsewhere (text editor).
- Be as concise and compact as possible. Make an effort of writing all you have to write at the time in a single block of text and posting it at once. Do not break a thought in multiple messages.
- React to what's being said with emojis to express agreement (:+1:), disagreement (:-1: ), confusion (:question:) and any other that efficiently describes your sentiment.

Please feel free to make any suggestions bellow.

**Comment by m52go** [comment]
## Meeting Minutes

Guiding question: 
**What are Bisq’s main pain points and what are the solutions being considered for it?**

Agenda:
```
A. It’s too costly to trade on Bisq because there are too many on-chain transactions 
and the mining fees are too high and will probably remain high.

Proposed solutions:
    1. Move to another protocol:
        1.1. ⚡ Lightning Network - this general idea keeps being mentioned and stejbac 
               made a proposal in February 2021 (https://github.com/bisq-network/proposals/issues/312)
        1.2. 💦 Liquid Network - wiz made a proposal on March 2020 to research the 
               implementation of Liquid Bitcoin as a second base currency on Bisq. The 
               proposal has been labeled as approved as of June 2020 (https://github.com/bisq-network/proposals/issues/198)

    2. Improve the use of the Bitcoin layer one
        2.1. sqrrm proposed to reduce the trade from four to a single on-chain transaction
        2.2. ManfredKarrer proposed in April 2019 a general concept of off-chain trading 
               with bonds as a security. The proposal has been labeled as stalled as of 
               August 2020 (https://github.com/bisq-network/proposals/issues/76)

B. It's too hard to use
    1. There's a need of good notifications app. The current developed by joachimneumann 
       hasn't been updated since January 2020
    2. Bisq API in the works - https://github.com/orgs/bisq-network/projects/17
```

### 1.1 ([off-chain trade protocol using lightning and bsq](https://github.com/bisq-network/proposals/issues/312))

There seemed to be a general sense of confusion and uncertainty about the proposal that only the author stejbac could dispel. Uncertainty revolved around the practical need to maintain a Lightning fork and tricky BSQ handling.

### 1.2 ([trade protocol using liquid as base currency](https://github.com/bisq-network/proposals/issues/198))

Seems like one of the more realistic paths forward at the moment, but there is a bit of unease at the concept of basing Bisq on a federated system instead of an open P2P network. More research is needed to evaluate.

Bisq will need to move off of BTC layer 1 but path forward is not clear. Seems certain Bisq will remain on layer 1 for at least 1 more year; after that (2-3 years on) it will probably be forced off, but it's unclear to what. Maybe another Bisq implementation that works as a proof-of-concept in parallel with layer-1 Bisq could be useful to test ideas and ease transition.

It's obvious the Bisq trade protocol will need to change dramatically in order for Bisq to remain relevant. Hopefully serious discussions taking place now can start to move ball in the right direction.

### 2.1 ([reduce trade protocol to a single on-chain transaction](https://github.com/bisq-network/proposals/issues/279))

Not much discussion about getting single-tx protocol to production, but:
- feedback and research indicate it should work
- main hurdle is development and testing resources
- current work on bsq-btc atomic swaps will add features that are needed for the single-tx protocol

Idea of a single-tx trade protocol is inspired by the way Lightning sets up pre-signed transactions as a fallback (i.e. none of them are broadcast if all goes well). Developer help for BSQ-BTC atomic swaps would be much appreciated, particularly on the UI side.

### 2.2 ([off-chain trade protocol with bsq bonding](https://github.com/bisq-network/proposals/issues/76))

Not likely to be happen any time soon...too many questions still unanswered. Also the idea of locking up large amounts of BSQ might make it unattractive for a lot of people.

### UI help

In addition to helping with BSQ-BTC atomic swaps UI, a JavaFX UI developer would be much appreciated to implement Pedro's new UI designs. 
- Resources to manage/oversee this new developer are available.
- Funding to trial this new developer has been volunteered.
- A promising developer was found a couple of months ago but his current status and willingness to work with Bisq is unknown.
- A sponsor for the full amount is still needed (assuming the developer refuses to accept BSQ for compensation).

### API release

Slated for the 1.6.0 release, but will only be accessible to those who build Bisq themselves, which is probably a good thing since it's probably not wise for anyone to start firing off commands just yet.

---

These meetings are slated to take place every 2 weeks for now, then will probably switch to every 4 weeks at some point.

**Reply by chimp1984** [reply]
> A promising developer was found a couple of months ago but his current status and willingness to work with Bisq is unknown.

He is still available but after repeated requests that stakeholders get in touch with him and nothing happened so far I am not sure how long he will stay available. Hope stakeholders can get a meeting with him soon.
